data_structures/mergesort/mergesort.py

```diff
@@ -0,0 +1,48 @@
+def mergesort(arr):
+    """
+    A recursive function that accepts an array of unsorted integers and
+    returns a sorted array
+    """
+    if type(arr) is not list:
+        raise TypeError('argument must be list of integers')
+
+    if len(arr) <= 1:
+        return arr    
+
+    def recurse(arr):
+        half = len(arr)//2
+        left = []
+        right = []
+
+        for i in range(half):
+            left = left + [arr[i]]
+            right = right + [arr[len(arr) - 1 - i]]
```

Comment:
Remember that you can use negative indexing to get items from the end of a list. This is equivalent to
```python
arr[-1 - i]
```